src/utils/validators.js

```diff
@@ -16,6 +16,8 @@ export const URL_REGEX_PREFIX = "^(https://)?(www.)?("
 export const URL_REGEX_SUFFIX = ".com/)([a-zA-Z0-9_-]+([/.])?)+$"
 export const TELEGRAM_REGEX = "telegram|t).me/([a-zA-Z0-9_-]+([/.])?)+$"
 export const TIKTOK_REGEX = ".com/@)([a-zA-Z0-9_-]+([/.])?)+$"
+export const DOMAIN_NAME_REGEX =
+  "(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]"
```

Comment:
we should add a lengthy comment here explaining what this regex does - this is p long and i'm not sure i understand it fully even w/ the help of regex101 (my lifesaver for regex ;--;)

also, does the initial `?` do anything? i was under the impression that it captures the previous group (of which there is none here)

could you also help me to understand teh `{0, 61}` - why do we have an upper bound here?

Reply:
I think that's a good idea, thanks! I have added a comment to explain each part of the regex plus some context behind some of the values. I also removed the `?:` part as it only differentiates between a capturing and non-capturing group, which is not relevant in this context.

For the `{0,61}`, it is part of RFC1035.

Reply:
thanks for doing this!